hivemind_redis_database/__init__.py

```
@@ -9,48 +9,343 @@
+            if old_client.name != client.name:
+                self.redis.srem(f"{self.index_prefix}:name:{old_client.name}", str(old_client.client_id))
+                self.redis.sadd(f"{self.index_prefix}:name:{client.name}", str(client.client_id))
+
+            if old_client.api_key != client.api_key:
+                self.redis.srem(f"{self.index_prefix}:api_key:{old_client.api_key}", str(old_client.client_id))
+                self.redis.sadd(f"{self.index_prefix}:api_key:{client.api_key}", str(client.client_id))
 
-            # Maintain indices for common search fields
-            self.redis.sadd(f"client:index:name:{client.name}", client.client_id)
-            self.redis.sadd(f"client:index:api_key:{client.api_key}", client.client_id)
             return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)